lldb/test/API/python_api/frame/TestFrames.py

```diff
@@ -91,7 +93,17 @@ def test_get_arg_vals_for_call_stack(self):
                     frame.GetSP(),
                     "SP gotten as a value should equal frame's GetSP",
                 )
-
+                # Test that the "register" property's flat list matches the list from
+                # the "registers" property that returns register sets:
+                register_regs = set()
+                flattened_regs = set()
+                for reg_set in frame.registers:
+                    for reg in reg_set:
+                        flattened_regs.add(reg.name)
+                for reg in frame.register:
+                    register_regs.add(reg.name)
```

Comment:
Instead of iterating over it twice, why not combine into one loop?

Reply:
Iterating over what twice?  One is iterating over the elements of the list that is the `registers` property, and the other using the generator for the `register` (note no final `s`) property.

`registers` is actually a poorly named property since it actually produces a list of `register sets`.

Reply:
Oh, I see now. I definitely did not notice the difference between `frame.register` and `frame.registers`... I wish we could change that. Maybe we could make `frame.register_sets` an alias for `frame.registers` and switch to using that?

Reply:
This PR is about fixing `register` not `registers`.  If you try to iterate over it in the current sources you get a weird error.  When I fixed just that error, then iterating over it spun forever, so I either had to explicitly turn off iterability or make some reasonable generator, which is what this patch does.  

Adding `register_sets` as another property that does the same thing as `registers` seems fine, but also not really a part of this PR.